azbus/sender.go

```diff
@@ -155,10 +155,9 @@ func (s *Sender) NewMessageBatch(ctx context.Context) (*OutMessageBatch, error)
 	return s.sender.NewMessageBatch(ctx, nil)
 }
 
-// BatchAddMessage calls Addmessage on batch
-// Note: this method is a direct pass through and exists only to provide a
-// mockable interface for adding messages to a batch.
-func (s *Sender) BatchAddMessage(batch *OutMessageBatch, m *OutMessage, options *azservicebus.AddMessageOptions) error {
+// BatchAddMessage calls Addmessage on batch and adds span tracing
+func (s *Sender) BatchAddMessage(ctx context.Context, span tracing.Span, batch *OutMessageBatch, m *OutMessage, options *azservicebus.AddMessageOptions) error {
+	s.updateSendingMesssageForSpan(ctx, m, span)
```

Comment:
you should be able to get the span from the ctx and not need the extra parameter?

Reply:
yes - but is already available so passing as an argument is more efficient

Reply:
At this point, the message is not sent, it is just added to the batch. The context passed in may not even be the context that is used for sending the batch. How is that going to work ?

And, secondly, does adding the span data to the context at this point record any information, like for example "start time", if it does it will not be representative of how batches are used in forestrie for certain. and probalby not in other places